app/controllers/reservations_controller.rb

```diff
@@ -0,0 +1,53 @@
+class ReservationsController < ApplicationController
+  before_action :authenticate_user!
+  before_action :require_donor
+
+
+  def ship
+    @reservation = Reservation.find(params[:id])
+
+    if @reservation.donor_id == current_user.id
+      @reservation.ship!
```

Comment:
sorry for unclear comment the other night - your call on which method to use here. `ship` will add validation errors to the model on failure, `ship!` will raise `StateMachines::InvalidTransition` on failure. we should be handling the failures accordingly either way.